src/shims/foreign_items.rs

```diff
@@ -744,6 +744,17 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     dest,
                 )?;
             }
+            "memcpy" => {
+                let [ptr_dest, ptr_src, n] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let ptr_dest = this.read_pointer(ptr_dest)?;
+                let ptr_src = this.read_pointer(ptr_src)?;
+                let n = this.read_target_usize(n)?;
+                let src_bytes =
+                    this.read_bytes_ptr_strip_provenance(ptr_src, Size::from_bytes(n))?.to_owned();
+                this.write_bytes_ptr(ptr_dest, src_bytes)?;
```

Comment:
We have a `mem_copy` method that does both of these at the same time https://doc.rust-lang.org/nightly/nightly-rustc/rustc_const_eval/interpret/eval_context/struct.InterpCx.html#method.mem_copy

I believe we should not strip provenance, but just copy everything over including provenance.